RadixWallet/Clients/HTTPClient/HTTPClient+Live.swift

```diff
@@ -3,14 +3,15 @@ extension HTTPClient {
 		let session = URLSession.shared
 
 		return .init(
-			executeRequest: { request in
+			executeRequest: { request, isStatusCodeValid in
 				let (data, urlResponse) = try await session.data(for: request)
 
 				guard let httpURLResponse = urlResponse as? HTTPURLResponse else {
 					throw ExpectedHTTPURLResponse()
 				}
 
-				guard httpURLResponse.statusCode == BadHTTPResponseCode.expected else {
+				let isSuccessStatusCode = isStatusCodeValid?(httpURLResponse.statusCode) ?? (httpURLResponse.statusCode == BadHTTPResponseCode.expected)
```

Comment:
not part of this PR, but wondering why are we checking the success code from something like `BadHTTPResponseCode` 😅

Reply:
Good question :) 
@GhenadieVP do you think we can move the expected success code elsewhere?

Reply:
I would say we can completely remove `expected` and `butExpected` from `BadHTTPResponseCode`